Unify <mark>'s UIA LocalizedControlType and AX AXRoleDescription

dandclark opened the pull request:
Per https://w3c.github.io/html-aam/#el-mark, the UIA LocalizedControlType for `<mark>` is "mark" while the AXRoleDescription is "highlighted content".  There is no good reason for these values to be different; this adds complexity for implementers and creates confusion for users when moving across platforms.  Furthermore, "mark" is a confusing name for non-programmers, and "highlighted content" is unnecessary verbose.

This change switches both role descriptions to "highlight", removing the inconsistency and improving upon both existing names.

Closes #236 

The following tasks have been completed:

 * [x] Confirmed there are no ReSpec/BikeShed errors or warnings.

Implementation commitment:

 * [x] WebKit (https://bugs.webkit.org/show_bug.cgi?id=201958)
 * [x] Chromium (https://bugs.chromium.org/p/chromium/issues/detail?id=996459)
* [x] Gecko (none yet)

CC: @aleventhal @cookiecrook  @boggydigital


<!--
    This comment and the below content is programatically generated.
    You may add a comma-separated list of anchors you'd like a
    direct link to below (e.g. #idl-serializers, #idl-sequence):

    Don't remove this comment or modify anything below this line.
    If you don't want a preview generated for this pull request,
    just replace the whole of this comment's content by "no preview"
    and remove what's below.
-->
***
<a href="https://pr-preview.s3.amazonaws.com/dandclark/html-aam/pull/235.html" title="Last updated on Sep 19, 2019, 1:09 AM UTC (6b2854d)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/html-aam/235/f7c394a...dandclark:6b2854d.html" title="Last updated on Sep 19, 2019, 1:09 AM UTC (6b2854d)">Diff</a>

dandclark committed:
Make <mark>'s UIA LocalizedControlType and AX AXRoleDescription consistent by changing both to 'highlight'

scottaohara commented:
if Webkit will commit to implement, then this looks like a good change to me.  Thank you

dandclark commented:
> if Webkit will commit to implement, then this looks like a good change to me.

@cookiecrook , is this something you'd be able to comment on?

gregwhitworth commented:
@cookiecrook Any update here? Thanks :)

cookiecrook commented:
Seems like a good change. 

WebKit bugzilla link above was broken, and I didn’t find an existing bug, so I filed a new one:

https://bugs.webkit.org/show_bug.cgi?id=201958

cookiecrook commented:
PS. Looks like the Gecko link is also broken.

dandclark commented:
I updated the link to the WebKit bug and removed the broken Gecko link ~~as this is N/A for them~~. [Correction: they'll still need to update their implementation but I assume the OK from WebKit is enough to move forward with this].
@scottaohara , is this PR good to complete now?
Thanks!

scottaohara commented:
thanks everyone
